storagemarket/impl/providerstates/provider_states.go

```diff
@@ -390,6 +392,15 @@ func HandoffDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal stor
 		log.Error(err)
 	}
 
+	// announce the deal to the network indexer
+	annCid, err := environment.AnnounceIndex(ctx.Context(), deal)
+	if err != nil {
+		log.Errorw("failed to announce index via reference provider", "proposalCid", deal.ProposalCid, "err", err)
```

Comment:
Should the fact that an error occurred in publish be saved to the deal state? Other than logs, how do I know which of my deals got announced successfully? Do I always want AnnounceAllDealsToIndex() to announce everything?

Reply:
@hannahhoward The next time a deal gets announced, because of the IPLD magic in the index provider, it will also announce all the previous un-announced deal when the network indexer asks for the latest.

Reply:
@aarshkshah1992 not necessarily. 
If the error happens because the advertisement failed to persist then that content will never get advertised. 
Looking at the code, such errors can hapen before call to `NotifyPut` e.g. in metadat encoding on markets side or inside the provider before persistence.

The case you mentioned above will only work if the new advertisement is added to the chain but not announced to the network due to say gossipsub error.

Reply:
https://github.com/filecoin-project/go-fil-markets/issues/685